test/listeners/listeners.module.spec.ts

```diff
@@ -0,0 +1,72 @@
+import {
+	LavalinkHostType,
+	LavalinkListenerMeta,
+	LavalinkListenersModule,
+	LavalinkListenerType
+} from '../../src';
+
+describe('LavalinkListenersModule', () => {
+	let module: LavalinkListenersModule;
+	let lavalinkManager: any;
+	let nodeManager: any;
+	let discoveryService: any;
+	let metadataScanner: any;
+	let reflector: any;
+
+	beforeEach(() => {
+		// Mocks
+		lavalinkManager = { on: jest.fn() };
+		nodeManager = { on: jest.fn() };
+
+		const fakeProviderInstance = {
+			onPlayerCreate: jest.fn()
+		};
+		discoveryService = {
+			getProviders: jest.fn().mockReturnValue([
+				{
+					instance: fakeProviderInstance,
+					isDependencyTreeStatic: jest.fn().mockReturnValue(true)
+				}
+			])
+		};
+
+		metadataScanner = {
+			getAllMethodNames: jest.fn().mockReturnValue(['onPlayerCreate'])
+		};
+
+		reflector = {
+			get: jest.fn().mockReturnValue({
+				event: 'playerCreate',
+				type: LavalinkListenerType.On,
+				host: LavalinkHostType.LavalinkManager
+			} as LavalinkListenerMeta)
+		};
+
+		module = new LavalinkListenersModule(
+			discoveryService,
+			metadataScanner,
+			reflector,
+			lavalinkManager,
+			nodeManager
+		);
+	});
+
+	it('should register LavalinkManager listener on onModuleInit', () => {
+		module.onModuleInit();
+
+		expect(discoveryService.getProviders).toHaveBeenCalled();
+		expect(metadataScanner.getAllMethodNames).toHaveBeenCalledWith(expect.any(Object));
+		expect(lavalinkManager.on).toHaveBeenCalledWith('playerCreate', expect.any(Function));
+	});
+
+	it('should call the decorated method when event is emitted', () => {
+		module.onModuleInit();
+
+		const callback = lavalinkManager.on.mock.calls[0][1];
```

Comment:
That makes the test brittle and harder to understand. Accessing to callback is not so clearly

```ts
  let registeredHandler: (...args: any[]) => void;

  lavalinkManager.on.mockImplementationOnce((event, handler) => {
    registeredHandler = handler;
  });
```